backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/VdsManager.java

```diff
@@ -544,6 +546,12 @@ public void updateUpdateAvailable(boolean updatesAvailable) {
     public void updateStatisticsData(VdsStatistics statisticsData) {
         vdsStatisticsDao.update(statisticsData);
         cachedVds.setStatisticsData(statisticsData);
+
+        statisticsData.getCpuCoreStatistics().stream().forEach(statistics -> {
+            cpuTopology.stream()
+                .filter(cpu -> cpu.getCore() == statistics.getCpuId())
```

Comment:
is it the right way to correlate that? I'd expect something that involved both thee core and the thread, no?

Reply:
We are just copying the usage from `CpuStatistics` object to `VdsCpuUnit` - both are identified by cpu id.

Reply:
```suggestion
                .filter(cpu -> cpu.getCpu() == statistics.getCpuId())
```
?

Reply:
Right, that was not correct of course.